### ISSUE TITLE
Corehost ouputs lots of `A JSON parsing exception occurred` errors with no diagnostic info

### ISSUE THREAD

**Issue by pakrym** [dotnet/cli#2115]
Looks like it gets exception parsing sdk version but error message itself is not helping at all. Doesn't show neither what error is, or what file it tries to parse.

```
Run target: build-compile
[09:05:01]Exec
[09:05:01]  program: dotnet
[09:05:01]  commandline: pack  src/Microsoft.Extensions.PlatformAbstractions --configuration Release
[09:05:01]  workingdir: /Users/asplab/buildAgent/work/169923259d6dcfec/Universe/PlatformAbstractions
[09:05:01]A JSON parsing exception occurred
[09:05:03]Compiling Microsoft.Extensions.PlatformAbstractions for .NETFramework,Version=v4.5.1
[09:05:03]A JSON parsing exception occurred
[09:05:10]
[09:05:10]Compilation succeeded.
[09:05:10]    0 Warning(s)
[09:05:10]    0 Error(s)
[09:05:10]
[09:05:10]Time elapsed 00:00:07.7260358
[09:05:10] 
[09:05:10]
[09:05:10]Compiling Microsoft.Extensions.PlatformAbstractions for .NETStandard,Version=v1.3
[09:05:10]A JSON parsing exception occurred
[09:05:16]
[09:05:16]Compilation succeeded.
[09:05:16]    0 Warning(s)
[09:05:16]    0 Error(s)
[09:05:16]
[09:05:16]Time elapsed 00:00:05.7872791
[09:05:16] 
```
/cc @schellap @davidfowl @pranavkm

**Comment by BillHiebert** [comment]
I am seeing the same thing from VS with 1.0.0.002117 when running dotnet build on a new web project. 

`
"C:\Program Files\dotnet\dotnet.exe" build "c:\Users\billhie\Documents\projects\WebApplication26\src\WebApplication26" --configuration Debug --no-dependencies --build-base-path "c:\Users\billhie\Documents\projects\WebApplication26\artifacts" --no-incremental
**A JSON parsing exception occurred**
Compiling WebApplication26 for .NETStandardApp,Version=v1.5
**A JSON parsing exception occurred**

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:03.0791203
 (The compilation time can be improved. Run "dotnet build --build-profile" for more information)
`

Another issue for VS, is that the json exceptions are being sent to stderr, so in addition to there being no information on the source of the error, VS fails the build as the msbuild task which calls dotnet build assumes messages sent to stderr are errors.

**Comment by schellap** [comment]
Is it possible to see what the line endings are in Windows?

**Comment by pakrym** [comment]
File name is still not displayed after latest fix.

**Comment by gkhanna79** [comment]
Is this the BOM issue?

**Comment by schellap** [comment]
Fixed, closing..

**Comment by schellap** [comment]
Fixed by #2186
